django/db/migrations/questioner.py

```diff
@@ -162,17 +163,16 @@ def ask_not_null_alteration(self, field_name, model_name):
         """Changing a NULL field to NOT NULL."""
         if not self.dry_run:
             choice = self._choice_input(
-                "You are trying to change the nullable field '%s' on %s to non-nullable "
-                "without a default; we can't do that (the database needs something to "
-                "populate existing rows).\n"
+                "It is impossible to change a nullable field '%s' on %s to non-nullable "
+                "without providing a default because the database needs something to "
+                "populate existing rows.\n"
                 "Please select a fix:" % (field_name, model_name),
                 [
-                    ("Provide a one-off default now (will be set on all existing "
-                     "rows with a null value for this column)"),
-                    ("Ignore for now, and let me handle existing rows with NULL myself "
-                     "(e.g. because you added a RunPython or RunSQL operation to handle "
-                     "NULL values in a previous data migration)"),
-                    "Quit, and let me add a default in models.py",
+                    ("Provide a one-off default now which will be set on all existing "
+                     "rows with a null value for this column."),
+                    ("Ignore for now (Note: existing rows that contain NULL values will have to be "
+                     "handled manually!)."),
```

Comment:
I don't think we need parentheses and `Note:`. Also, there is no need to use `!`. Maybe:
```suggestion
                    'Ignore for now. Existing rows that contain NULL values '
                    'will have to be handled manually, for example with a '
                    'RunPython or RunSQL operation.',
```
```

Reply:
Hey @felixxm, I have a question about string quotations. I am grasping to understand when are single vs double quotes used in Django repository. So far, the only pattern I could notice is that one-line strings contain single quotes and breaking lines always contain double quotes. However, this suggestion uses single quotes in a breaking sentence. 
Can you please explain what is the best practice?

Thanks!

Reply:
For new code, we're using single quotes (unless a string contains a single quote) as described in [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).